### ISSUE TITLE
Redis support

### ISSUE THREAD

**Issue by stefankleff** [juriansluiman/SlmQueue#153]
Is anybody already working on that? 

We're currently switching our cache backend to redis and I like to use that as queueing backend too. But I also like to stick with SlmQueue. If no work is done yet, I can start to write that. But I'm not yet very familiar with Redis. So I need to glimpse at the php-resque(-ex) and Laravel implementations and may need some help as well as a review.

**Comment by bakura10** [comment]
Hi,

No work has been done, but maybe @juriansluiman can open a SlmQueueRedis ? :)

**Comment by juriansluiman** [comment]
No, no work has been done yet but a while ago I gave it some thought. @stefankleff if you want we can collaborate on a Redis backend for SlmQueue.

A few things worthy I have thought about:

1. Check [the slides](http://www.slideshare.net/rosstuck/redis-for-the-everyday-developer/154) from Ross Tuck about Redis and esp. from slide 154 and on about queues.
2. Redis has a PHP extension (fast!) and a php library (slower, installable via composer). We should support both via an adapter & interface, so you can swap those around
3. Like we do with SQL, you need a "pending" queue and a "running" queue, with `BRPOPLPUSH` you can atomically switch jobs from queue "pending" to queue "running". It's blocking, so it waits for a new job to come by automatically

**Comment by stefankleff** [comment]
Sounds good! I hope that I can start working on that within next week

**Comment by juriansluiman** [comment]
See juriansluiman/SlmQueueRedis#1 for the issue I opened to get started. I have made a small outline, but if you have alternative ideas you can completely ignore those.

**Comment by roelvanduijnhoven** [comment]
@stefankleff I don't think the Redis adapter is going anywhere right ;).

**Comment by stefankleff** [comment]
Not really. I'm working now in another company and where it is not used.
